src/EntryPoints/PersistentPageIdentifiersHooks.php

```diff
@@ -18,12 +18,14 @@ class PersistentPageIdentifiersHooks {
 	public static function onInfoAction( IContextSource $context, array &$pageInfo ): void {
 		$pageInfo['header-basic'][] = [
 			$context->msg( 'persistentpageidentifiers-info-label' ),
-			PersistentPageIdentifiersExtension::getInstance()->newPersistentPageIdFormatter()->format(
-				self::getPersistentIdForPage( $context->getWikiPage() )
-			)
+			htmlspecialchars( self::formatPersistentId( self::getPersistentIdForPage( $context->getWikiPage() ) ) )
```

Comment:
If escaping is needed here (looks plausible), then doesn't the above message also need escaping?

I don't remember if `$context->msg` escapes but suspect not, because there is some `->escaped()` you can call on it.

I also don't know what `$pageInfo['header-basic']` expects. Does the code that uses it do escaping close to the output, or does it expect already escaped values? You can test by hardcoing some HTML here and seeing if it shows up in the UI or if it gets escaped.

Reply:
Value escaping is necessary here. I tested with HTML in the format initially, that's why I added it.

The message gets escaped automatically.

This is the code in the PR (+ hardcoded HTML in message):
![Screenshot_20241115_171918](https://github.com/user-attachments/assets/fd1ed71f-1229-4f3b-815a-7e85bb547113)

And this is without `htmlspecialchars`:
![Screenshot_20241115_172022](https://github.com/user-attachments/assets/f7de5918-7fab-4984-83d8-846f56e0fe8e)